lib/redis/connection/ruby.rb

```diff
@@ -27,8 +27,12 @@ module SocketMixin
       CRLF = "\r\n".freeze
 
       # Exceptions raised during non-blocking I/O ops that require retrying the op
-      NBIO_EXCEPTIONS = [Errno::EWOULDBLOCK, Errno::EAGAIN]
-      NBIO_EXCEPTIONS << IO::WaitReadable if RUBY_VERSION >= "1.9.3"
+      NBIO_READ_EXCEPTIONS = [Errno::EWOULDBLOCK, Errno::EAGAIN]
+      NBIO_WRITE_EXCEPTIONS = [Errno::EWOULDBLOCK, Errno::EAGAIN]
+      if RUBY_VERSION >= "1.9.3"
```

Comment:
Sidebar: any plans for deprecating Ruby 1.8 support? It's been EOL for 2 years now...